datasets/cmu_mocap/README.md

```diff
@@ -0,0 +1,307 @@
+        reader = c3d.Reader(open(motion, 'rb'))
+        for i, points, analog in reader.read_frames():
+            print('frame {}: point {}, analog {}'.format(i, points.shape, analog.shape))
+    
+    ...
+    ...
+cmu_mocap = cmu_mocap.map(process_motions)
+
+```
+
+
+- `avi` and `mpg`: AVI (Audio Video Interleave) has been the long-standing format to save and deliver movies and other video files. An MPG file is a common video file that uses a digital video format standardized by the Moving Picture Experts Group (MPEG)
+
+You can use a video loader library such as [decord](https://github.com/dmlc/decord) to load and process these files.
+
+Here is some sample preprocessing code for the `avi` format, similarly can be used for `mpg` as well.
+```python
+from datasets import load_dataset
+from decord import VideoReader
+ 
+cmu_mocap = load_dataset("./datasets/cmu_mocap", "avi")
+ 
+def process_motions(ex):
+    motions = ex['motions']
+    # Using the decord library
+    for motion in motions:
+        vr = VideoReader(motion)
+        print("Video frames: ", len(vr))
+        ...
+    ...
+    ...
+
+cmu_mocap = cmu_mocap.map(process_motions)
+```
+
+### Supported Tasks and Leaderboards
+
+[More Information Needed]
+
+### Languages
+
+All the categories, sub-categories and descriptions use English as their primary language.
+
+## Dataset Structure
+
+### Data Instances
+
+A sample from the CMU MoCap dataset is provided below:
+```
+{
+    'subject_id': 12, 
+    'categories': ['Physical Activities & Sports', 'Locomotion', 'Locomotion', 'Locomotion'], 
+    'subcategories': ['martial arts', 'walking', 'walking', 'walking'], 
+    'descriptions': ['tai chi', 'walk', 'walk', 'walk'], 
+    'motions': {
+        'amc_files': ['~/all_asfamc/subjects/12/12_04.amc', '~/all_asfamc/subjects/12/12_01.amc', '~/all_asfamc/subjects/12/12_03.amc', '~/all_asfamc/subjects/12/12_02.amc'], 
+        'skeleton_file': '~/all_asfamc/subjects/12/12.asf',
+    }
+}
+```
+
+### Data Fields
+
+The following fields are common for all file formats of the dataset.
+
+- `subject_id`: Unique ID for each subject in the MoCap dataset.
+- `categories`:  A list containing the category names of all trials.
+- `subcategories`: A list containing the subcategory names of all trials.
+- `descriptions`: A list containing the descriptions of all trials.
+
+The subject's motion trials in the `asf-amc` format are represented as follows:
+- `motions`:
+  - `asf`: A string containing the path to the ASF skeleton file.
+  - `amc`:  A list containing the paths to all AMC motion files.
+
+In the `c3d`, `mpg` and `avi` formats, this field has the following structure:-
+- `motions`: A list containing the path to all motion files.
+
+
+### Data Splits
+
+All the data is contained in training set. 
+
+Not all subjects were converted from the initial `asf-amc` format to the destination formats, so depending on the format, the number of examples varies:
+<MARKDOWN TABLE>
```

Comment:
What I meant by `<MARKDOWN TABLE>` is a markdown table with the size of each dataset (a table similar to [this one](https://huggingface.co/datasets/imdb#data-splits)) should be placed here.